starport-core/src/main/scala/com/krux/starport/metric/CloudWatchReporter.scala

```diff
@@ -0,0 +1,886 @@
+package com.krux.starport.metric
+
+import java.util._
+import java.util.concurrent.{ConcurrentHashMap, Future, TimeUnit}
+
+import scala.collection.JavaConverters._
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchAsync
+import com.amazonaws.services.cloudwatch.model.{
+  Dimension,
+  MetricDatum,
+  PutMetricDataRequest,
+  PutMetricDataResult,
+  StandardUnit,
+  StatisticSet
+}
+import com.codahale.metrics.{
+  Clock,
+  Counter,
+  Counting,
+  Gauge,
+  Histogram,
+  Meter,
+  Metered,
+  MetricFilter,
+  MetricRegistry,
+  ScheduledReporter,
+  Snapshot,
+  Timer
+}
+import org.slf4j.{Logger, LoggerFactory}
+import com.krux.starport.metric.CloudWatchReporter._
+
+/**
+ * Reports metrics to <a href="http://aws.amazon.com/cloudwatch/">Amazon's CloudWatch</a> periodically.
+ * <p>
+ * Use {@link CloudWatchReporter.Builder} to construct instances of this class. The {@link CloudWatchReporter.Builder}
+ * allows to configure what aggregated metrics will be reported as a single {@link MetricDatum} to CloudWatch.
+ * <p>
+ * There are a bunch of {@code with*} methods that provide a sufficient fine-grained control over what metrics
+ * should be reported
+ */
+class CloudWatchReporter(builder: Builder)
```

Comment:
let's also make the constructor private e.g. `class CloudWatchReporter private (builder: Builder)` to make it clear that builder pattern should be used